api/src/main/java/com/epam/esm/controller/advice/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,275 @@
+   * @return IncorrectData object containing original error message and custom error code
+   */
+  @ExceptionHandler(ConstraintViolationException.class)
+  @ResponseStatus(HttpStatus.BAD_REQUEST)
+  public IncorrectData handleException(
+      ConstraintViolationException constraintViolationException, HttpServletRequest request) {
+    String errorMessage = getLocalizedMessage(constraintViolationException, request);
+    return new IncorrectData(constraintViolationException, errorMessage);
+  }
+
+  /**
+   * Prevents from being sent the data for beans that does not match validation criteria.
+   *
+   * @param methodArgumentNotValidException the instance of MethodArgumentNotValidException
+   * @return IncorrectData object containing original error message and custom error code
+   */
+  @ExceptionHandler(MethodArgumentNotValidException.class)
+  @ResponseStatus(HttpStatus.BAD_REQUEST)
+  public IncorrectData handleException(
+      MethodArgumentNotValidException methodArgumentNotValidException, HttpServletRequest request) {
+    String errorMessage = getLocalizedMessage(methodArgumentNotValidException, request);
+    return new IncorrectData(methodArgumentNotValidException, errorMessage);
+  }
+
+  /**
+   * Prevents from sending to the datasource the value that duplicates existing entry in the unique
+   * field.
+   *
+   * @param sqlIntegrityConstraintViolationException the instance of
+   *     SQLIntegrityConstraintViolationException
+   * @return IncorrectData object containing original error message and custom error code
+   */
+  @ExceptionHandler(SQLIntegrityConstraintViolationException.class)
+  @ResponseStatus(HttpStatus.BAD_REQUEST)
+  public IncorrectData handleException(
+      SQLIntegrityConstraintViolationException sqlIntegrityConstraintViolationException,
+      HttpServletRequest request) {
+    String errorMessage = getLocalizedMessage(sqlIntegrityConstraintViolationException, request);
+    return new IncorrectData(sqlIntegrityConstraintViolationException, errorMessage);
+  }
+
+  /**
+   * Prevents from being sent incorrect format body content.
+   *
+   * @param httpMessageNotReadableException the instance of HttpMessageNotReadableException
+   * @return IncorrectData object containing original error message and custom error code
+   */
+  @ExceptionHandler(HttpMessageNotReadableException.class)
+  @ResponseStatus(HttpStatus.BAD_REQUEST)
+  public IncorrectData handleException(
+      HttpMessageNotReadableException httpMessageNotReadableException, HttpServletRequest request) {
+    String messageKey = "message.http_message_not_readable_exception";
+    String errorMessage = getLocalisedMessageFromBundle(messageKey, getLocale(request));
+    return new IncorrectData(httpMessageNotReadableException, errorMessage);
+  }
+
+  private Locale getLocale(HttpServletRequest request) {
+    String header = request.getHeader(HttpHeaders.ACCEPT_LANGUAGE);
+    Locale locale;
+    if (header == null || header.isBlank()) {
```

Comment:
You can replace this with Optional.ofNullable with filter + map

Reply:
Done!